File: src/0058-length-of-last-word/solution.go
package solution

import (
	"strings"
)

/**
 * trim 掉字符串末尾所有空格
 */
func lengthOfLastWord(s string) int {
	s = strings.TrimRight(s, " ")
	if s == "" {
		return 0
	}
	sList := strings.Split(s, " ")
	return len(sList[len(sList)-1])
}

File: src/0058-length-of-last-word/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    // 去掉最右空格
    s = s.replace(/(\s*$)/g, '');
    if (s == '') {
        return 0;
    }
    var s_list = s.split(' ');
    return s_list[s_list.length - 1].length;
}
